tikv/split_region.go

```diff
@@ -141,80 +142,57 @@ func (s *KVStore) batchSendSingleRegion(bo *Backoffer, batch kvrpc.Batch, scatte
 		}
 	}
 
-	req := tikvrpc.NewRequest(tikvrpc.CmdSplitRegion, &kvrpcpb.SplitRegionRequest{
-		SplitKeys: batch.Keys,
-	}, kvrpcpb.Context{
-		Priority: kvrpcpb.CommandPri_Normal,
-	})
+	opts := make([]pd.RegionsOption, 0, 1)
+	if tableID != nil {
+		opts = append(opts, pd.WithGroup(fmt.Sprintf("%v", *tableID)))
+	}
 
-	sender := locate.NewRegionRequestSender(s.regionCache, s.GetTiKVClient())
-	resp, err := sender.SendReq(bo, req, batch.RegionID, client.ReadTimeoutShort)
+	// Split regions by pd
+	spResp, err := s.pdClient.SplitRegions(bo.GetCtx(), batch.Keys, opts...)
 
-	batchResp := kvrpc.BatchResult{Response: resp}
-	if err != nil {
-		batchResp.Error = errors.Trace(err)
-		return batchResp
-	}
-	regionErr, err := resp.GetRegionError()
+	resp := SplitResponse{splitResp: spResp}
 	if err != nil {
-		batchResp.Error = errors.Trace(err)
-		return batchResp
-	}
-	if regionErr != nil {
-		err := bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
-		if err != nil {
-			batchResp.Error = errors.Trace(err)
-			return batchResp
-		}
-		resp, err = s.splitBatchRegionsReq(bo, batch.Keys, scatter, tableID)
-		batchResp.Response = resp
-		batchResp.Error = err
-		return batchResp
+		resp.Error = err
+		return resp
 	}
 
-	spResp := resp.Resp.(*kvrpcpb.SplitRegionResponse)
-	regions := spResp.GetRegions()
-	if len(regions) > 0 {
+	regionsId := spResp.RegionsId
+	if len(regionsId) > 0 {
 		// Divide a region into n, one of them may not need to be scattered,
 		// so n-1 needs to be scattered to other stores.
-		spResp.Regions = regions[:len(regions)-1]
-	}
-	var newRegionLeft string
-	if len(spResp.Regions) > 0 {
-		newRegionLeft = logutil.Hex(spResp.Regions[0]).String()
+		spResp.RegionsId = regionsId[:len(regionsId)-1]
```

Comment:
We should also check `finished_percentage` in response, only if the `percentage` become 100 we will think the splitting is success.